cmake/Dependencies.cmake

```diff
@@ -1428,6 +1428,14 @@ if(USE_DISTRIBUTED AND USE_TENSORPIPE)
       set(TP_USE_CUDA ON CACHE BOOL "" FORCE)
       set(TP_ENABLE_CUDA_IPC ON CACHE BOOL "" FORCE)
     endif()
+    if(USE_ROCM)
+      add_compile_options(-D__HIP_PLATFORM_HCC__=1)
```

Comment:
Why are we still using `__HIP_PLATFORM_HCC__`?  It is long deprecated at this point.  If this is needed for the tensorpipe code, we should also update the tensorpipe code to instead use `USE_ROCM` just like pytorch does today.

Reply:
I removed `__HIP_PLATFORM_HCC__`.
For the tensorpipe there is a variable `TP_USE_ROCM` defined to separate ROCm specific code.

Reply:
HIP needed the variable - `__HIP_PLATFORM_AMD__` so added it.

9:39:16 Nov 13 03:39:16 In file included from /var/lib/jenkins/pytorch/third_party/tensorpipe/tensorpipe/common/hip_buffer.h:12,
19:39:16 Nov 13 03:39:16                  from /var/lib/jenkins/pytorch/third_party/tensorpipe/tensorpipe/common/hip_buffer.cc:10:
19:39:16 Nov 13 03:39:16 /opt/rocm/include/hip/hip_runtime.h:66:2: error: #error ("Must define exactly one of __HIP_PLATFORM_AMD__ or __HIP_PLATFORM_NVIDIA__");
19:39:16 Nov 13 03:39:16    66 | #error("Must define exactly one of __HIP_PLATFORM_AMD__ or __HIP_PLATFORM_NVIDIA__");
19:39:16 Nov 13 03:39:16       |  ^~~~~
19:39:16 Nov 13 03:39:16 In file included from /opt/rocm/include/hip/hip_runtime.h:113,